packages/mongodb-mql-model/src/main/kotlin/com/mongodb/jbplugin/mql/Node.kt

```diff
@@ -110,6 +117,26 @@ data class Node<S>(
     override fun toString(): String {
         return "Node(components=$components)"
     }
+
+    fun queryHash(): Int {
```

Comment:
It seems that two nodes might not be equivalent depending on the "pass" during analysing. This was affecting the Inlay, as it couldn't use the cached results of the index verification from other run processes like the inspections. This simple query hash is used in the ExplainQuery slice to have a consistent hash based on the referenced fields in a query.